tap_github/streams.py

```diff
@@ -266,6 +266,112 @@ class CommunityProfileStream(GitHubStream):
     ).to_dict()
 
 
+class EventsStream(GitHubStream):
+    """
+    Defines 'Events' stream.
+    Issue events are fetched from the repository level (as opposed to per issue)
+    to optimize for API quota usage.
+    """
+
+    name = "events"
+    path = "/repos/{org}/{repo}/events"
+    primary_keys = ["id"]
+    replication_key = "created_at"
+    parent_stream_type = RepositoryStream
+    state_partitioning_keys = ["repo", "org"]
+    ignore_parent_replication_key = False
+
+    def get_records(self, context: Optional[dict] = None) -> Iterable[Dict[str, Any]]:
+        """Return a generator of row-type dictionary objects.
+        Each row emitted should be a dictionary of property names to their values.
+        """
+        if context and context.get("events", None) == 0:
+            self.logger.debug(f"No events detected. Skipping '{self.name}' sync.")
+            return []
+
+        return super().get_records(context)
+
+    def post_process(self, row: dict, context: Optional[dict] = None) -> dict:
+        # TODO - We should think about the best approach to handle this. An alternative would be to do
+        # a 'dumb' tap that just keeps the same schemas as GitHub without renaming these objects to "target_".
+        row["target_repo"] = row.pop("repo", None)
+        row["target_org"] = row.pop("org", None)
+        return row
```

Comment:
In doing a final pass, I came across this block and remembered in another PR we discussed there may be upcoming collisions on these two fields.

Are these redundant by chance with the `repo` and `org` string fields we are using for paritioning? Is there much to lose by removing them and just downstream, getting any additional needed properties like id/login/name from the parent stream using those parent context keys?

Reply:
Sadly, when I checked these could be different. Esp for forks and merge events between forks and parents...

Reply:
I added some details in the comment for now